src/nwb_benchmarks/benchmarks/ibl_chunking.py

```diff
@@ -0,0 +1,100 @@
+"""Basic benchmarks for stream NWB files and their contents."""
+
+import parameterized
+
+from ..core import (
+    get_s3_url,
+    read_hdf5_fsspec_no_cache,
+    read_hdf5_nwbfile_fsspec_no_cache,
+    read_hdf5_nwbfile_remfile,
+    read_hdf5_nwbfile_ros3,
+    read_hdf5_remfile,
+    read_hdf5_ros3,
+)
+
+ibl_chunk_experiments = [
+    dict(
+        dandiset_id="000717",
+        dandi_path="sub-IBL-ecephys/sub-IBL-ecephys_ses-3e7ae7c0_desc-18000000-frames-13653-by-384-chunking.nwb",
+    )
+]
+
+
+@parameterized.parameterized_class(
+    [
+        dict(
+            dandiset_id="000717",
+            dandi_path="sub-IBL-ecephys/sub-IBL-ecephys_ses-3e7ae7c0_desc-18000000-frames-13653-by-384-chunking.nwb",
+        )
+    ]
+)
+class FsspecNoCacheHDF5FileReadBenchmark:
+    repeat = 1
+
+    def setup(self):
+        self.s3_url = get_s3_url(dandiset_id=self.dand, dandi_path=self.dandi_path)
+
+    def time_file_read(self):
+        self.nwbfile, self.io, self.file, self.bytestream = read_hdf5_fsspec_no_cache(s3_url=self.s3_url)
```

Comment:
Are `self.nwbfile, self.io, self.file, self.bytestream` actually being used? If not, should we just say:

```suggestion
        _ = read_hdf5_fsspec_no_cache(s3_url=self.s3_url)
```

Reply:
In general I'm looking to both mimic the exact way a user would call the function (including assigning all intermediate steps to variables) and avoid timing garbage collection as a part of the operation (which I think would trigger here if the the variables aren't assigned and the reference count goes to zero)

Reply:
It might be splitting hairs, but I've seen (and written) somewhat complicated `__del__`/`__exit__` conditions that might otherwise trigger

Reply:
Makes sense. I was not thinking about garbage collection. Might be worth adding a note in the "writing tests" part of the docs/readme